students/Zach Cooper/lesson07/parallel.py

```diff
@@ -0,0 +1,226 @@
+#                   %(message)s"
+#     logging.basicConfig(level=logging.WARNING, format=log_format,
+#                         filename='mylog.log')
+
+
+def csv_list(file):
+    """
+        Iterates through csv file and file contents in a list
+        param: csvfile
+    """
+    return_list = []
+    with open('sample_csv_files/' + file, encoding='utf-8-sig') as csv_file:
+        file_input = csv.DictReader(csv_file)
+        for row in file_input:
+            return_list.append(row)
+    return return_list
+
+
+def show_available_products(db):
+    """
+        Return a dictionary with each product listed infor
+        :param db: MongoDB
+        :retun in a dict: product_id, description, product_type, quantity_available
+    """
+    mongo = MongoDBConnection()
+
+    with mongo:
+        # mongodb database connects here
+            db = mongo.connection.HB_Norton_DB
+
+    products_available = {}
+
+    for product in db.products.find():
+        if int(product["quantity_available"]) > 0:
+
+            products_dict = {"description": product["description"],
+                             "product_type": product["product_type"],
+                             "quantity_available": product['quantity_available']}
+
+            products_available[product["product_id"]] = products_dict
+    return products_available
+
+
+def show_rentals(db, product_id):
+    """
+        Return information from user who have rented products on product_id
+        Used MongoDB aggregate() framework methods I found online
+        : param db: MongoDB
+        : param product_id: proudctid
+        : return: user_id, name, address, phone_number, email
+    """
+    mongo = MongoDBConnection()
+
+    with mongo:
+        db = mongo.connection.HB_Norton_DB
+
+        # Aggretion framework used in MongoDB
+        output_list = list(db.rentals.aggregate([
+            {'$match':
+                {'product_id': product_id}},
+            {'$lookup':
+                {'from': "customers",
+                 'localField': "user_id",
+                 'foreignField': "user_id",
+                 'as': "cust"}},
+            {'$project':
+                {'_id': 0,
+                 'user_id': 1,
+                 'name': '$cust.name',
+                 'address': '$cust.address',
+                 'phone_number': '$cust.phone_number',
+                 'email': '$cust.email'}}]))
+        return(output_list)
+
+
+def drop_db_data(collection):
+    """
+        Deletes the collections from MongoDB
+        :param db: MongoDB
+        :return: MongoDB empty with no collection names
+    """
+    mongo = MongoDBConnection()
+
+    with mongo:
+        db = mongo.connection.HB_Norton_DB
+        db[collection].drop()
+        return True
+
+
+class db_thread(Thread):
+    """
+        Create subclass that inputs csv contents through multiple threads
+        Queue import helps info be exhchanged safely between multiple threads.
+
```

Comment:
A subclass is still a class, and should be capitalized.  DBThread would be fine.